docs/topics/multiplatform/multiplatform-intro.md

```diff
@@ -0,0 +1,102 @@
+[//]: # (title: Introduction to Kotlin Multiplatform)
+
+Support for multiplatform programming is one of Kotlin's key benefits. It reduces time spent writing and maintaining the
+same code for [different platforms](multiplatform-dsl-reference.md#targets) while retaining the flexibility and benefits of native programming.
+
+![Kotlin Multiplatform](kotlin-multiplatform.svg){width=700}
+
+## Learn key concepts
+
+Kotlin Multiplatform allows you to share code across different platforms, whether it's mobile, web, or desktop.
+The platforms to which the code is compiled are defined by the list of _targets_.
+
+Each target has a corresponding *source set*, which represents a set of source files with its own dependencies and
+compiler options. Platform-specific source sets, for example `jvmMain` for the JVM, can make use of platform-specific
+libraries and APIs.
+
+To share code among a subset of targets, intermediate source sets are used. For example, the `appleMain` source set
+represents the code shared among all Apple platforms. The code shared among all platforms and compiled to all declared
+targets has its own source set, `commonMain`. It cannot use platform-specific APIs, but can take advantage of
+multiplatform libraries.
+
+When compiling for a specific target, Kotlin combines the common source set, the relevant intermediate source sets,
+and the target-specific source set.
+
+For more details on this topic, see:
+
+* [The basics of Kotlin Multiplatform project structure](multiplatform-discover-project.md)
+* [Set up targets manually](multiplatform-set-up-targets.md)
+* [Advanced concepts of the multiplatform project structure](multiplatform-advanced-project-structure.md)
+
+## Use code sharing mechanisms
+
+It's sometimes more convenient to share code among a subset of similar targets. Kotlin Multiplatform provides a way to
+simplify their creation with a *default hierarchy template*. It includes a pre-defined list of intermediate source sets
+that are created based on the targets you specified in your project.
+
+To access platform-specific APIs from shared code, you can use another Kotlin mechanism, *expected and actual declarations*.
+This way, you can declare that you `expect` a platform-specific API in common code but provide a separate `actual`
+implementation for each target platform. You can use this mechanism with different Kotlin concepts, including functions,
```

Comment:
```suggestion
implementation for each target platform. You can use this mechanism with functions,
```

Is there anything besides functions, classes, and interfaces? "Including" makes it sound like there's something else that can be expected.

Reply:
Well, technically yes, it could also be enums, properties, annotations